### PR TITLE
Support for named package categories

### PR DESCRIPTION
**Pull request by matteius**
In support of: https://github.com/pypa/pipenv/pull/5366

### CODE REVIEW

File: src/requirementslib/models/project.py [hunk 1]
```diff
@@ -70,155 +66,3 @@ def dumps(self):
         strio = io.StringIO()
         self.model.dump(strio)
         return strio.getvalue()
-
```
Comment (oz123) [inline src/requirementslib/models/project.py:73]: The Lockfile class has references to Project files which should be cleaned too.

**Review by oz123 — CHANGES_REQUESTED** [review]

File: src/requirementslib/models/project.py (hunk 1 above)
Comment (matteius) [reply src/requirementslib/models/project.py:73]: @oz123 `ProjectFile` is still a thing as its used in some capacity, its `Project` that I removed -- I don't see reference to `Project` in lockfile.py ...

Comment (oz123) [reply src/requirementslib/models/project.py:73]: Ok. Sorry. I misread it.

**Review by oz123 — APPROVED** [review]

**Comment by oz123** [comment]
Err? What is this license compliance?

**Comment by matteius** [comment]
> Err? What is this license compliance?

It is a step that always fail but we lack permission into that system to see why it is.

**Pull request merged**